kubernetes/v1alpha1/spec.go

```diff
@@ -58,11 +58,17 @@ type MonitoringDashboardChart struct {
 	Min          *int                             `json:"min"`
 	Max          *int                             `json:"max"`
 	MetricName   string                           `json:"metricName"`
+	Metrics      []MonitoringDashboardMetric      `json:"metrics"`
```

Comment:
Now that this library it's part of production versions like ServiceMesh 1.0 and ServiceMesh 1.1.
Have we thought if these changes may have some impact in the operator work when upgrading things ?

I'm inclined to think that not, it's fine.
A SM 1.1 that upgrades to SM 1.2 will reset it's specification and that's fine.

But I just leave this question here in case there are some other aspect that I missed.

In case of yes, perhaps it would need some process to upgrade one version to another.

In general I haven't seen a strong use of v1alpha1 -> v1alpha2 etc, I don't want to handle this versioning for this.
It's overarch, but I think that review about if operator should be able to update the CRDs from SM 1.1 to SM 1.2 (which may include this) and what should do with existing CRs (perhaps a patch).

I don't expect an action on the PR about this, but perhaps an Issue to not forget and describing any additional action on that side.

Reply:
Same as you, I don't see any reason why it wouldn't work, I hope I'm not missing something. The fact that we don't change the CRD should precisely make the things easier / without problem, I think.